test/framework/utilities.py

```diff
@@ -51,6 +51,30 @@
 from easybuild.tools.filetools import mkdir, read_file
 from easybuild.tools.module_naming_scheme import GENERAL_CLASS
 from easybuild.tools.modules import modules_tool
+from easybuild.tools.options import EasyBuildOptions
+
+
+# make sure tests are robust against any non-default configuration settings;
+# involves ignoring any existing configuration files that are picked up, and cleaning the environment
+# this is tackled here rather than in suite.py, to make sure this is also done when test modules are ran separately
```

Comment:
This will make other tests which rely on `$EASYBUILD_X` (without `TEST`) that are defined by the test itself, or which use temporary configuration files that are put in place deliberately, needlessly complicated...

I'm really not in favor of monkey patching unless it's absolutely necessary (e.g., to mock certain functionality), the tests should test 'real' functionality, not stuff that has been tampered with to make testing 'easier'.

This block of code effectively shields off the tests from any configuration options that were set via available configuration files or environment variables, and it's done such that existing tests are not affected.